### PR TITLE
meta key option when returning multiple models

### LINKED ISSUE #8: Proposal · Provide different set of keys used for meta tags
Proposal --

If returning multiple items from a route, allow to return a model under the key `meta`.

So for example...
```
currentRouteModel.meta.articleTitle || currentRouteModel.articleTitle
```

### PR DESCRIPTION
**Pull request by snewcomer**
close #8

The is first stab at allowing users to pass a `meta` key, especially in cases where one would return multiple models from the route model hook.

### CODE REVIEW

File: addon/services/head-data.js [hunk 1]
```diff
@@ -11,8 +11,14 @@ export default Service.extend({
...
   routeName: alias('routing.currentRouteName'),
 
-  currentRouteModel: computed('routeName', function() {
-    return getOwner(this).lookup(`route:${this.get('routeName')}`).get('currentModel');
+  currentRoute: computed('routeName', function() {
+    return getOwner(this).lookup(`route:${this.get('routeName')}`);
+  }),
+
+  currentRouteModel: computed('currentRoute', function() {
+    let metaAlias = this.get('currentRoute.metaAlias');
```
Comment (snewcomer) [inline addon/services/head-data.js]: not sure if this is the best name.  It is meant to indicate that there will be multiple models returned from the model hook and you should pick _this_ one...

File: tests/acceptance/meta-test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  });
+
+  test('coming from anothe route', async function(assert) {
+    await visit('/blog/post');
+    await visit('/meta');
+
+    assert.equal(currentURL(), '/meta');
+    assert.equal(document.head.querySelector('meta[name="twitter:title"]').content,
+      'Overridden Meta', 'twitter title is correct');
+  });
```
Comment (snewcomer) [inline tests/acceptance/meta-test.js]: this test seems useful...coming from another route.

**Comment by snewcomer** [comment]
Also, does this work with the afterModel hook?  i.e.

```
model() {
  return this.store.findRecord('post');
},
afterModel(post) {
  return {
    post: post
    meta: {
      title: post.get('name')
    }
  }
}
```

Or would we want to shy away from this pattern and think about a better way to implement with something like #9?

File: addon/services/head-data.js (hunk 1 above)
Comment (rwwagner90) [reply addon/services/head-data.js]: Perhaps `metaKey` or `modelNameForMeta` or something along those lines? I'm not sure. @mansona any opinions on naming?

File: tests/acceptance/meta-test.js (hunk 2 above)
Comment (rwwagner90) [reply tests/acceptance/meta-test.js]: Definitely 👍

**Comment by rwwagner90** [comment]
I used to update my models in the `afterModel` hook to add or massage meta, but I did it something like:

```js
afterModel(model) {
  model.title = 'Override Title';
}
```

Does returning an object from `afterModel` replace your model?

**Comment by mansona** [comment]
I don't quite understand the benefit of this over just overriding the `currentRouteModel()` hook 🤔

**Comment by rwwagner90** [comment]
@snewcomer @mansona I am already overriding it in my head-data service, like so:
```js
  currentRouteModel: computed('routeName', function() {
    return getOwner(this).lookup(`route:${this.get('routeName')}`).get('currentModel.attributes');
  }),
```

So I think doing some sort of API to override on a per route basis could be nice. I'm not too picky on the actual implementation though. Whatever you guys think, is probably fine with me 😃

**Comment by rwwagner90** [comment]
@snewcomer @mansona what do you guys think of this approach? https://github.com/shipshapecode/ember-meta/pull/13

I believe allowing to define a `metaInfo` property on the route, rather than using models, addresses these issues, correct?

**Pull request closed without merging**